tiledb/sm/compressors/util/gzip_wrappers.cc

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file gzip_wrappers.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2022 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines gzip_compress and gzip_uncompress, routines
+ * that slightly simplify use of the gzip compression routines.
+ */
+
+#include "gzip_wrappers.h"
+
+#include <inttypes.h>
+
+Status gzip_compress(
+    shared_ptr<tiledb::sm::Buffer>& out_gzipped_buf,
+    const void* in_bytes,
+    uint64_t nbytes) {
+  /**
+   * buffer format:
+   *    uint64_t uncompressed_size
+   *    uint64_t compressed_size
+   *    uint8_t [compressed_size] - compressed data
+   */
+  uint64_t overhead_size = 2 * sizeof(uint64_t);
+  tiledb::sm::ConstBuffer const_in_buf(in_bytes, nbytes);
+  // Ensure space in output buffer for worst acceptable case.
+  if (!out_gzipped_buf->realloc(nbytes + overhead_size).ok()) {
+    return LOG_STATUS(
+        Status_CompressionError("gzip output buffer allocation error"));
+  }
+  out_gzipped_buf->reset_offset();
+  out_gzipped_buf->reset_size();
+  tiledb::sm::Buffer* out_buffer_ptr = out_gzipped_buf.get();
+  assert(out_buffer_ptr != nullptr);
+  out_buffer_ptr->reset_offset();
+  out_buffer_ptr->advance_size(overhead_size);
+  out_buffer_ptr->advance_offset(overhead_size);
+  printf("overhead_size %" PRIu64 "\n", overhead_size);
+  if (!tiledb::sm::GZip::compress(9, &const_in_buf, out_buffer_ptr).ok()) {

```

Comment:
Don't really see a reason to have `out_buffer_ptr`, can use `out_gzipped_buf.get()`here.